### ISSUE TITLE
[feature] import / export config

### ISSUE THREAD

**Issue by jrpie** [jrpie/Launcher#79]
(suggested by Sarp Küçük)

#  Please describe the problem to be solved

When moving to a new device, manually restoring the old settings is tedious.

#  Describe the solution you would suggest

There should be an option to export the settings and import them on the new device

**Comment by jrpie** [comment]
depends on #81

**Comment by jrpie** [comment]
There are some limitations:
* It is not clear how to map users ids from one device to user ids of another device. Probably it would be best to store the [user type](https://source.android.com/docs/devices/admin/multi-user#user_types) of all users when exporting the config and then try to find a user of the same type when importing.
* It is impossible to restore pinned shortcuts (the other app must initiate pinning the shortcut itself)
* For the same reason it is impossible to restore widgets (see #44, which is not yet implemented)

**Comment by wassupluke** [comment]
Perhaps this [documentation](https://commonsware.com/Room/pages/chap-backup-001) from a comment on this [thread](https://stackoverflow.com/questions/78698538/how-can-i-implement-a-feature-to-export-and-import-database-to-my-app-in-android#comment138751836_78698538) helps?

**Comment by jrpie** [comment]
> Perhaps this [documentation](https://commonsware.com/Room/pages/chap-backup-001) on `androidx.room` from a comment on this [thread](https://stackoverflow.com/questions/78698538/how-can-i-implement-a-feature-to-export-and-import-database-to-my-app-in-android#comment138751836_78698538) helps?


Not really. The problem is not how to save / read data but how to restore it, in particular:

(a) how to deal with data exported by an older version (e.g. look at the mess in [de.jrpie.android.launcher.preferences.legacy](https://github.com/jrpie/launcher/tree/master/app/src/main/java/de/jrpie/android/launcher/preferences/legacy) which is only going to get worse; maybe I'll drop support for very old versions when releasing 1.0)
(b) how to try to restore stuff like widgets
(c) how to match user profiles from one device with user profiles on another device

I know what I have to do, it is just a lot of work to actually do it 😅 


> Is this a 0.3.0 milestone?

Sound like a good idea
